.github/workflows/build-devmode.yaml

```diff
@@ -0,0 +1,56 @@
+name: Build the dev mode UI
+
+on:
+  # Runs on pushes targeting the `main` branch, and only for the `frontend/dev-mode/**` path.
+  pull_request:
+    paths:
+      - 'frontend/dev-mode/**'
+
+  # Allows us to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+env:
+  PNPM_CACHE_STORE_PATH: ""
+
+jobs:
+  build-ui:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
```

Comment:
These are older versions of the actions... We also pin the actions to their SHA values and let dependabot do the updates. See platform-relese workflow which uses the same actions for the current versions in use.